stingray/events.py

```diff
@@ -0,0 +1,212 @@
+        ----------------
+        dt: float
+            The time resolution of the events. Only relevant when using events
+            to produce light curves with similar bin time.
+
+        energies: iterable
+            A list of array of photon energy values
+
+        mjdref : float
+            The MJD used as a reference for the time array.
+
+        gtis: [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+            Good Time Intervals
+
+        pi : integer, numpy.ndarray
+            PI channels
+
+        pha : float, numpy.ndarray
+            Photon energies
+
+        Attributes
+        ----------
+        time: numpy.ndarray
+            The array of event arrival times, in seconds from the reference
+            MJD (self.mjdref)
+
+        energies: numpy.ndarray
+            The array of photon energy values
+
+        ncounts: int
+            The number of data points in the event list
+
+        dt: float
+            The time resolution of the events. Only relevant when using events
+            to produce light curves with similar bin time.
+
+        mjdref : float
+            The MJD used as a reference for the time array.
+
+        gtis: [[gti0_0, gti0_1], [gti1_0, gti1_1], ...]
+            Good Time Intervals
+
+        pi : integer, numpy.ndarray
+            PI channels
+
+        pha : float, numpy.ndarray
+            Photon energies
+
+        """
+        self.time = np.array(times, dtype=np.longdouble)
+        self.energy = np.array(energies)
+        self.ncounts = len(times)
+        self.notes = notes
+        self.dt = dt
+        self.mjdref = mjdref
+        self.gti = gti
+        self.pi = pi
+        self.pha = pha
+
+    @staticmethod
+    def from_fits(fname, **kwargs):
+        from .utils import _assign_value_if_none
+        from .io import load_events_and_gtis
+
+        ret = load_events_and_gtis(fname, **kwargs)
+
+        times = ret.ev_list
+        gti = ret.gtis
+
+        pi = _assign_value_if_none(ret.additional_data, "PI", None)
+        pha = _assign_value_if_none(ret.additional_data, "PHA", None)
+        dt = ret.dt
+        mjdref = ret.mjdref
+
+        return EventList(times, gti=gti, pi=pi, pha=pha, dt=dt, mjdref=mjdref)
+
+    def to_lc(self, bin_time, start_time=None, stop_time=None, center_time=True):
+        """
+        Convert event list to a light curve object.
+
+        Parameters
+        ----------
+        bin_time : float
+            Binning time of the light curve
+
+        Other Parameters
+        ----------------
+        start_time : float
+            Initial time of the light curve
+        stop_time : float
+            Stop time of the light curve
+        center_time: bool
+            If False, time is the start of the bin. Otherwise, the center
+
+        Returns
+        -------
+        lc: `Lightcurve` object
+        """
+
+        times, counts = events_to_lc(self.time, bin_time, start_time,
+            stop_time, center_time)
+
```

Comment:
`events_to_lc` duplicates functionality in the static method `Lightcurve.makeLightcurve`, which you could just return instead of doing these two steps.